### PR TITLE
Aggregator rpc

### PR DESCRIPTION
**Pull request by hexoscott**

### CODE REVIEW

File: cmd/rpcdaemon/commands/zkevm_api.go [hunk 1]
```diff
@@ -43,6 +52,7 @@ type ZkEvmAPI interface {
 	GetWitness(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash, debug *bool) (hexutility.Bytes, error)
 	GetBlockRangeWitness(ctx context.Context, startBlockNrOrHash rpc.BlockNumberOrHash, endBlockNrOrHash rpc.BlockNumberOrHash, debug *bool) (hexutility.Bytes, error)
 	GetBatchWitness(ctx context.Context, batchNumber uint64) (hexutility.Bytes, error)
+	GetAggregatorDetail(ctx context.Context, batchNumber uint64, debug *bool) (*legacy_executor_verifier.RpcPayload, error)
```
Comment (mandrigin) [inline cmd/rpcdaemon/commands/zkevm_api.go]: GetProverInput

File: cmd/rpcdaemon/commands/zkevm_api.go [hunk 2]
```diff
@@ -382,6 +400,96 @@ func (api *ZkEvmAPIImpl) GetBatchWitness(ctx context.Context, batchNumber uint64
...
+	oldAccInputHash, err := api.l1Syncer.GetOldAccInputHash(ctx, &api.config.L1PolygonRollupManager, ApiRollupId, batchNumber)
+	if err != nil {
+		return nil, err
+	}
+
+	timestampLimit := lastBlock.Time()
+
+	return &legacy_executor_verifier.RpcPayload{
+		Witness:           hex.EncodeToHex(rangeWitness),
+		DataStream:        hex.EncodeToHex(streamBytes),
```
Comment (mandrigin) [inline cmd/rpcdaemon/commands/zkevm_api.go]: I am thinking that that one is not really necessary, they can get the datastream from our datastream server.

File: cmd/rpcdaemon/commands/zkevm_api.go [hunk 3]
```diff
@@ -382,6 +400,96 @@ func (api *ZkEvmAPIImpl) GetBatchWitness(ctx context.Context, batchNumber uint64
...
+		Coinbase:          api.config.SequencerAddress.String(),
+		OldAccInputHash:   oldAccInputHash.String(),
+		L1InfoRoot:        "", // todo [zkevm] this might be needed in the future - would need to track l1 events
```
Comment (mandrigin) [inline cmd/rpcdaemon/commands/zkevm_api.go]: let's remove it for now

**Review by mandrigin — DISMISSED** [review]

**Comment by hexoscott** [comment]
all points actioned now

**Review by mandrigin — APPROVED** [review]

**Pull request merged**